pkg/api/componentreadiness/component_report.go

```diff
@@ -1594,120 +1649,179 @@ func (c *componentReportGenerator) assessComponentStatus(requiredConfidence, sam
 		baseTotal = baseFailure + baseSuccess + baseFlake
 	}
 
-	status := crtype.MissingBasis
-	testStats := crtype.ReportTestStats{
-		SampleStats: crtype.TestDetailsReleaseStats{
-			Release: c.SampleRelease.Release,
-			TestDetailsTestStats: crtype.TestDetailsTestStats{
-				SuccessRate:  getSuccessRate(sampleSuccess, sampleFailure, sampleFlake),
-				SuccessCount: sampleSuccess,
-				FailureCount: sampleFailure,
-				FlakeCount:   sampleFlake,
-			},
-		},
-		BaseStats: crtype.TestDetailsReleaseStats{
+	if baseTotal == 0 && c.RequestAdvancedOptions.PassRateRequiredNewTests > 0 {
```

Comment:
Core logic here, I tried to push almost everything into assessComponentStatus, and broke out fishers vs pass rate functionality into separate functions.